ARSClient.mjs

```diff
@@ -205,80 +181,73 @@ class ARSClient
         running: running,
         errored: errored
       };
-    }
-    else
-    {
+    } else {
       let agents = Object.keys(completed);
       // Get uuids corresp. to these agents, fetch their results in parallel
       let toFetch = agents.map(e => completed[e].uuid);
-      const promises = toFetch.map(async (e) =>
-        {
-          console.log(`kicking off fetch for ${e}`);
-          return this.fetchMessage(e)
-        });
+      let start = new Date();
+      const promises = toFetch.map(async (e) => {
+        console.log(`kicking off fetch for ${e}`);
+        return this.fetchMessage(e);
+      });
       let finalCompleted = [];
       // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#parameters
-      await Promise.allSettled(promises).then(results =>
-        {
-          results.forEach(item =>
-            {
-              if (item.status === 'fulfilled')
-              {
-                let agent = item.value.fields.name;
-                let elem = completed[agent];
-                elem.data = item.value.fields.data.message;
-                finalCompleted.push(elem);
-                console.log(`settled ${agent}`);
-              }
-              else
-              {
-                //
-                console.error('Unexpected case of being unable to fetch a result for an agent that reported code=200');
-                errored.push(item.value); // No idea what might be in this object
-              }
-            });
-          console.log('done settling promises');
-          retval = {
-            pk: pkey,
-            completed: finalCompleted,
-            running: running,
-            errored: errored
-          };
+      await Promise.allSettled(promises).then(results => {
+        results.forEach(item => {
+          if (item.status === 'fulfilled') {
+            let itemData = item.value[1];
+            let agent = itemData.fields.name;
+            let elem = completed[agent];
+            elem.data = itemData.fields.data.message;
+            elem.meta = item.value[0];
+            finalCompleted.push(elem);
+            console.log(`settled ${agent}`);
+          } else {
+            //
+            console.error('Unexpected case of being unable to fetch a result for an agent that reported code=200');
+            errored.push(item.value); // No idea what might be in this object
+          }
         });
+        console.log('done settling promises');
+        retval = {
+          pk: pkey,
+          completed: finalCompleted,
+          running: running,
+          errored: errored
+        };
+      });
+      meta.childrenProcessingMs = new Date() - start;
     }
     retval.queuing = false;
+    retval.meta = meta;
     return retval;
   }
 
   // Get all results that have been pre-merged by the ARS
-  async collectMergedResults(pkey, statusCheck = false)
-  {
+  // *=* TODO: Need to update w/ new metadata-having interface!
```

Comment:
Should do this now to avoid fumbling around when we switch back to ARS merging (soon?)